src/backend/distributed/commands/utility_hook.c

```diff
@@ -616,11 +666,88 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		if (isAlterExtensionUpdateCitusStmt)
 		{
 			citusCanBeUpdatedToAvailableVersion = !InstalledAndAvailableVersionsSame();
+
+			/*upgrade citus: alter extension citus update to 'xxx' */
```

Comment:
would be good to move the new code into a separate function

Reply:
done